torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -1184,14 +1160,23 @@ def _pad_with_scalar_fill(
     fill: Union[int, float],
     padding_mode: str,
 ) -> torch.Tensor:
-    shape = image.shape
-    num_channels, height, width = shape[-3:]
+    old_height, old_width = image.shape[-2:]
```

Comment:
I've put the handling only in `_pad_with_scalar_fill` since that is the only thing that we can hit for masks:

https://github.com/pytorch/vision/blob/4af683107f967d4d435be4180020989dd8c3019c/torchvision/transforms/v2/functional/_geometry.py#L1268-L1269

However, we could also move it up to `pad_image`. LMK if you prefer that.